mgmtsystem/azure.py

```diff
@@ -166,11 +166,18 @@ def delete_vm(self, vm_name, resource_group=None):
         vhd_name = os.path.split(urlparse.urlparse(vhd_endpoint).path)[1]
         self.remove_blob_image(vhd_name)
 
-    def list_vm(self):
+    def list_vm(self, resource_group=None):
+        """Returns a list of VM Names inside a particular resource group.
+
+        Args:
+            resource_group - Uses value if passed, otherwise uses default.
+        """
         self.logger.info("Attempting to List Azure VMs")
-        azure_data = self.run_script("Get-AzureRmVm | convertto-xml -as String")
-        data = self.clean_azure_xml(azure_data)
-        vm_list = etree.parse(StringIO(data)).getroot().xpath(
+        azure_data = self.run_script(
+            """
+                Get-AzureRmVm -ResourceGroupName \"{rg}\" | convertto-xml -as String")
```

Comment:
It looks like there's a trailing double quote here after String, or am I seeing things.